staging/src/k8s.io/client-go/metadata/metadatainformer/informer.go

```diff
@@ -31,6 +32,33 @@ import (
 	"k8s.io/client-go/tools/cache"
 )
 
+// SharedInformerOption defines the functional option type for metadataSharedInformerFactory.
+type SharedInformerOption func(*metadataSharedInformerFactory) *metadataSharedInformerFactory
+
+// WithTweakListOptions sets a custom filter on all listers of the configured metadataSharedInformerFactory.
+func WithTweakListOptions(tweakListOptions TweakListOptionsFunc) SharedInformerOption {
+	return func(factory *metadataSharedInformerFactory) *metadataSharedInformerFactory {
+		factory.tweakListOptions = tweakListOptions
+		return factory
+	}
+}
+
+// WithNamespace limits the metadataSharedInformerFactory to the specified namespace.
+func WithNamespace(namespace string) SharedInformerOption {
+	return func(factory *metadataSharedInformerFactory) *metadataSharedInformerFactory {
+		factory.namespace = namespace
+		return factory
+	}
+}
```

Comment:
How is this related?

Reply:
No, this is not related. But I'd like to make metadataSharedInformerFactory have identical SharedInformerOption(s) with SharedInformerFactory for consistency.